### PR TITLE
perf: add deployAadManifest interface 

### PR DESCRIPTION
**Pull request by wenytang-ms**
the error msg string:
https://msazure.visualstudio.com/Microsoft%20Teams%20Extensibility/_sprints/taskboard/AuthAndData/Microsoft%20Teams%20Extensibility/CY22-10.2?workitem=15553175

### CODE REVIEW

**Review by SLdragon — APPROVED** [review]

File: packages/fx-core/src/core/FxCore.ts [hunk 1]
```diff
@@ -268,6 +272,51 @@ export class FxCore implements v3.ICore {
...
+    EnvInfoLoaderMW_V3(false),
+    ContextInjectorMW,
+    EnvInfoWriterMW_V3(),
+  ])
+  async deployAadManifest(inputs: Inputs, ctx?: CoreHookContext): Promise<Result<Void, FxError>> {
+    setCurrentStage(Stage.deployAad);
+    inputs.stage = Stage.deployAad;
+    const updateAadClient = Container.get("aadApp/update") as any;
+    // In V3, the aad.template.json exist at .fx folder, and output to root build folder.
+    const manifestTemplatePath: string = path.join(inputs.projectPath!, ".fx", "aad.template.json");
```
Comment (blackchoey) [inline packages/fx-core/src/core/FxCore.ts:292]: a heads up that `aad.template.json` will be moved out of `.fx` folder

**Review by blackchoey — DISMISSED** [review]

**Review by SLdragon — DISMISSED** [review]

File: packages/fx-core/src/core/FxCore.ts (hunk 1 above)
Comment (wenytang-ms) [reply packages/fx-core/src/core/FxCore.ts:292]: Will update the path after settled

**Review by blackchoey — APPROVED** [review]

File: packages/fx-core/tests/common/local/localSettingsHelper.test.ts [hunk 2]
```diff
@@ -153,7 +153,7 @@ describe("localSettingsHelper", () => {
     let projectSettings: ProjectSettingsV3;
 
     beforeEach(() => {
-      sinon.stub(process, "env").value({ TEAMSFX_V3: "true" });
+      sinon.stub(process, "env").value({ TEAMSFX_API_V3: "true" });
```
Comment (blackchoey) [inline packages/fx-core/tests/common/local/localSettingsHelper.test.ts:156]: why change to `TEAMSFX_API_V3`?

**Review by blackchoey — APPROVED** [review]

**Review by jayzhang — APPROVED** [review]

File: packages/vscode-extension/package.json [hunk 3]
```diff
@@ -616,8 +616,7 @@
       },
       {
         "command": "fx-extension.deployAadAppManifestFromCtxMenu",
-        "title": "%teamstoolkit.commands.deployAadAppManifestFromCtxMenu.title%",
-        "enablement": "fx-extension.isAadManifestEnabled"
```
Comment (HuihuiWu-Microsoft) [inline packages/vscode-extension/package.json:620]: Any reason to remove this?

Comment (wenytang-ms) [reply packages/vscode-extension/package.json:620]: this is by design, no need to check the condition, and user can trigger this command anytime.
we check the condition in fx-core, and throw error on output

**Review by HuihuiWu-Microsoft — APPROVED** [review]

**Pull request merged**